src/selectors/entities/channels.js

```diff
@@ -306,20 +306,33 @@ export const getMembersInCurrentChannel = createSelector(
 export const getUnreads = createSelector(
     getAllChannels,
     getMyChannelMemberships,
-    (channels, myMembers) => {
+    getUsers,
+    getCurrentUserId,
+    (channels, myMembers, users, currentUserId) => {
         let messageCount = 0;
         let mentionCount = 0;
         Object.keys(myMembers).forEach((channelId) => {
             const channel = channels[channelId];
             const m = myMembers[channelId];
+
             if (channel && m) {
+                let otherUserId;
                 if (channel.type === 'D') {
-                    mentionCount += channel.total_msg_count - m.msg_count;
+                    otherUserId = getUserIdFromChannelName(currentUserId, channel.name);
+                    if (users[otherUserId]) {
```

Comment:
don't you need to also check if the other user is not deleted? you may have it in the store cause it was previously loaded or something...

Reply:
>don't you need to also check if the other user is not deleted

I didn't get you which `other user` you are referring to? . DM is between two users so, I am checking for the user other than the currentUser here.

Reagrding store:
Web: Deleted user will be part of redux store till a refreash. User will continue to see unread undicator and deleted user in LHS so, they can visit the DM and it clear automatically or they can refreash in this case deleted user will no longer be available in LHS and in store along with decreased unread count as it is supposed to be.

Mobile: Same case with mobile but here it is till clear cache.

Reply:
Sorry I needed to make this clearer, other user means the second user in the DM that is not the current user.

Just for the fact that you need to refresh or clear the cache means that the user might still be in the redux state and the unread counts will include messages from this user, so you need to check that the user is in the store and deleted_at equals to 0

Reply:
> other user means the second user in the DM that is not the current user.

Yes, i am referring to the other user(Not the current one). We only look for other user data in store.

> Just for the fact that you need to refresh or clear the cache means that the user might still be in the redux state and the unread counts will include messages.

Yes, Please read the above comment once again. Deleted user is not removed from the store immediately. So till a refreash or clear cache you will see the user in LHS. But i dont see how this messes up with unread Logic.

If user is present in redux store or in cache you will see them in LHS and unread count appropriately i.e (Unread count)  so i dont see a descripanacy because it adds up to the count.

After refreash or clear cache unread count still adds up because `otherUser` is not part of LHS or in cache so - (unread count - deleted users count)

Regarding `deleted_at` i think you meant to say check for !== 0 but this probably is a bug in CLI as deleted_at is always zero for DM channels objects.

Reply:
ok I can see what you mean..

2 things:
1. needs rebase
2. should we use the release-5.2 branch for this?

Reply:
I don't know if we should consider it for 5.2 as this is not a regression and it touches unread logic.

If this should be considered then i can rebase and raise it against 5.2

Reply:
> Regarding deleted_at i think you meant to say check for !== 0 but this probably is a bug in CLI as deleted_at is always zero for DM channels objects.

I believed it's not to DM channel but to the other user, like:
```javascript
if (users[otherUserId] && users[otherUserId].delete_at > 0)
```

Reply:
`but to the other user, like:` are you saying this needs to be added for channel?

I noticed one bug when testing this. For a just archived channel with unreads the count is not in sync with LHS. I will fix this and push a commit.

Reply:
Oh no, sorry.  I was thinking something like: 
```javascript
if (users[otherUserId] && users[otherUserId].delete_at === 0)
```